### PR TITLE
Add rank correlation metrics

### PR DESCRIPTION
**Pull request by kzkadc**
Description: added Spearman and Kendall rank correlation coefficient metrics.

https://en.wikipedia.org/wiki/Spearman%27s_rank_correlation_coefficient
https://en.wikipedia.org/wiki/Kendall_rank_correlation_coefficient

Check list:

- [x] New tests are added (if a new feature is added)
- [x] New doc strings: description and/or example code are in RST format
- [x] Documentation is updated (if required)

### CODE REVIEW

File: ignite/metrics/regression/kendall_correlation.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
+from typing import Any, Callable, Tuple
+
+import torch
+
+from scipy.stats import kendalltau
```
Comment (vfdev-5) [inline ignite/metrics/regression/kendall_correlation.py:5]: Let's put this import into `_compute_kendall_tau` method and in the constructor of `KendallRankCorrelation` let's check if scipy is present in the environment and raise an error if not like here:
https://github.com/pytorch/ignite/blob/aa3e3e13c214fe6cf72e941a46f13378911c8894/ignite/metrics/cohen_kappa.py#L64-L67

File: ignite/metrics/regression/spearman_correlation.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
+from typing import Any, Callable, Tuple
+
+import torch
+
+from scipy.stats import spearmanr
```
Comment (vfdev-5) [inline ignite/metrics/regression/spearman_correlation.py:5]: Same comment for this import like for kendall corr implementation

**Review by vfdev-5 — COMMENTED** [review]
Thanks for the PR @kzkadc , I left few comments now about how to handle the imports of scipy in the code. I'll check the tests and docs a bit later, but overall implementation looks ok to me.

File: ignite/metrics/regression/kendall_correlation.py (hunk 1 above)
Comment (kzkadc) [reply ignite/metrics/regression/kendall_correlation.py]: Thank you for your review!
I addressed that.

File: ignite/metrics/regression/kendall_correlation.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+            0.4666666666666666
+    """
+
+    def __init__(
+        self,
+        variant: str = "b",
+        output_transform: Callable[..., Any] = lambda x: x,
+        check_compute_fn: bool = True,
+        device: str | torch.device = torch.device("cpu"),
```
Comment (vfdev-5) [inline ignite/metrics/regression/kendall_correlation.py:89]: Docs build is failed due to the pipe in the type annotation, let's use Union. We still support python 3.8 and 3.9 before their EOL.

**Comment by vfdev-5** [comment]
Please also check code formatting, unit tests CI is also failed due to too long line

**Review by vfdev-5 — APPROVED** [review]
LGTM, thanks for the PR @kzkadc !

**Comment by vfdev-5** [comment]
CI failures are unrelated, submitted a temporary fix with mlflow version constraints. Once https://github.com/pytorch/ignite/pull/3278 is merged let's update this PR, rerun the CI and merge if all is green.

**Pull request merged**